#! /bin/sh /usr/share/dpatch/dpatch-run
## fix_large_ranges_in_random_generator_r146995.dpatch by Mirco Bauer <meebey@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: 2009-11-26  Sebastien Pouliot  <sebastien@ximian.com>
## DP:	* Random.cs: Fix case where the diff between min and max
## DP:    is very large in Next(int,int). Fix bug #558593
## DP:	  [Backport r146993]

@DPATCH@
diff -urNad git~/mcs/class/corlib/System/Random.cs git/mcs/class/corlib/System/Random.cs
--- git~/mcs/class/corlib/System/Random.cs	2009-10-26 21:44:45.000000000 +0100
+++ git/mcs/class/corlib/System/Random.cs	2009-12-13 19:49:47.000000000 +0100
@@ -124,7 +124,7 @@
 			if (diff <= 1)
 				return minValue;
 
-			return (int)(Sample () * diff) + minValue;
+			return (int)((uint)(Sample () * diff) + minValue);
 		}
 
 		public virtual void NextBytes (byte [] buffer)
